python/pyspark/sql/connect/column.py

```diff
@@ -161,7 +161,7 @@ def to_plan(self, session: Optional["RemoteSparkSession"]) -> proto.Expression:
         return self.ref.to_plan(session)
 
 
-class ScalarFunctionExpression(Expression):
+class ScalarFunctionExpression(ColumnRef):
```

Comment:
Why a function expression inherit from ColumnRef now?

Reply:
I'll refactor this to make the difference between Expression and ColumnRef clear. Essentially, I'll move all of the binary operators up to the Expression class.